### PR TITLE
[3593][16.0][ADD] stock_lot_label_layout_custom

### PR DESCRIPTION
**Pull request by AungKoKoLin1997**
[3593](https://www.quartile.co/web#id=3593&cids=3&menu_id=505&action=1457&model=project.task&view_type=form)

### CODE REVIEW

**Comment by AungKoKoLin1997** [comment]
@rinaldifirdaus Can you please review this first? I use the size as you suggested in task but it looks like not perfect.

**Comment by rinaldifirdaus** [comment]
Hi Lin, i've added 1 commit to adjust font size, padding of the barcode and add expiry date same with the default template.
this is still need to be improved, i compare the sample i created before with the current pdf result, seems like this module version having a letter spacing on it, also the left-border of the barcode is cutted.. could you please help to check this lin?
image

**Comment by AungKoKoLin1997** [comment]
Current report will be like below.
[Lot-Serial - 11111 (2).pdf](https://github.com/qrtl/rco-custom/files/12377562/Lot-Serial.-.11111.2.pdf)

**Review by rinaldifirdaus — APPROVED** [review]
Printed the barcode, and the result now same with the sample. thanks lin!

File: report_lot_custom_layout/wizards/stock_lot_label_layout.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import logging
+
+from odoo import fields, models
+
+
+class LotLabelLayout(models.TransientModel):
+    _inherit = "lot.label.layout"
+
+    print_format = fields.Selection(
+        selection_add=[("A4", "A4")], ondelete={"A4": "cascade"}
```
Comment (yostashiro) [inline report_lot_custom_layout/wizards/stock_lot_label_layout.py]: I suppose this option should be selected by default.

File: report_lot_custom_layout/wizards/stock_lot_label_layout.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    )
+
+    def process(self):
+        report_action = super(LotLabelLayout, self).process()
+        # Replace xml_id or modify the result
+        if self.print_format == "A4":
+            logging.info("Testing__________")
+            logging.info(report_action)
```
Comment (yostashiro) [inline report_lot_custom_layout/wizards/stock_lot_label_layout.py]: Do we need these logging lines?

File: report_lot_custom_layout/readme/DESCRIPTION.rst [hunk 3]
```diff
@@ -0,0 +1 @@
+This module adds new custom report for lot label report.
```
Comment (yostashiro) [inline report_lot_custom_layout/readme/DESCRIPTION.rst]: @rinaldifirdaus Please update the description with background information.

**Review by yostashiro — COMMENTED** [review]
Name suggestion; stock_lot_label_layout_custom

File: stock_lot_label_layout_custom/reports/stock_lot_barcode.xml
Comment (yostashiro) [inline stock_lot_label_layout_custom/reports/stock_lot_barcode.xml:1]: Would be good to reduce the node laddars by creating templates for some tag blocks (reduce the laddars as a general guideline).  Not a blocking point.

**Review by yostashiro — COMMENTED** [review]
@rinaldifirdaus Please regenarate README as it doesn't show the latest info.

**Comment by rinaldifirdaus** [comment]
@yostashiro san, i have updated the readme of this module.
as per discussed, we will not reduce the node laddars for now.
thank you

**Pull request merged**